### ISSUE TITLE
Producer fails to start silently

### ISSUE THREAD

**Issue by samuelgmartinez** [awslabs/amazon-kinesis-producer#26]
I had a problem with transitive dependencies requiring different versions of the Jackson library. I had an incorrect POM definition that included the older Jackson version, so when the auth chain calls the ProfileCredentialsProvider the call crashed due to a LinkageError because the provider uses a non existent method in that old version. So far it's an expected behavior.

What is not expected is the producer behaves like it were working correctly, even when the daemon is not even running. Moreover, is the library doesn't generate any error logging at all. We had to enable remote debug do discover that.

The problem is the try & catch in the [Daemon constructor](https://github.com/awslabs/amazon-kinesis-producer/blob/master/java/amazon-kinesis-producer/src/main/java/com/amazonaws/services/kinesis/producer/Daemon.java#L123). I'm not saying that the constructor should catch the error, since Error classes should bubble up by definition, but there should be a way to inform the client code that the daemon failed to start using a exception or something, don't you think? I mean, why the fork is happening inside an executor and not in the current thread? Or at least, it should check the future to get the exception if that is the case. Maybe I'm missing something regarding cross-platform compatibility or something like that.

With the current implementation the client code doesn't have any method to know that the shit hit the fan so the app can't act according to it :-1:

**Comment by lc-nyovchev** [comment]
I have to agree here. The blackbox that is the native daemon should have better and more relevant error messages and not swallow errors.

**Comment by pfifer** [comment]
Thanks for reporting this.  With the way that the KPL starts up it's not really possible to throw an exception on construction.  It should be possible to throw an exception on addUserRecord.  Would that provide the failure information you need?
